extensions/funqy/funqy-server-common/deployment/src/main/java/io/quarkus/funqy/deployment/FunctionScannerBuildStep.java

```diff
@@ -120,8 +123,12 @@ public void transform(TransformationContext transformationContext) {
                 ClassInfo clazz = transformationContext.getTarget().asClass();
                 if (!classes.contains(clazz))
                     return;
-                Transformation transformation = transformationContext.transform()
-                        .add(BuiltinScope.DEPENDENT.getName());
+                if (BuiltinScope.isDeclaredOn(clazz)) {
+                    // nothing to do as the presence of a scope will automatically qualify the class as a bean
+                    return;
```

Comment:
You don't want to add the `@Typed` annotation in this case?

Reply:
Nope, because Arc will do it automatically